make/cpp2asm_support.cmake

```diff
@@ -334,6 +334,19 @@ elseif (UNIX)
 else ()
   # Even if we didn't preprocess we'd need our own rule since cmake doesn't
   # support ml.
+  # UDPDATE: it does via enable_language(ASM_MASM) but as asm files have C/C++ commens in
+  #          preprocessing is required, all asm files are renamed to <name>.obj.s for later
+  #          compiling, which will make default CMAKE_ASM_COMPILER to fail with unknown extension
+  #          when using nmake, solution is to replace it with proper ml/ml64 before, or properly use
+  #          find_program in case of NMake Makefiles
```

Comment:
I'm not sure I understand: the code up at lines 229-236 does a find_program for ml64 (or ml) and gives a fatal error if not found.  So wouldn't we have hit a fatal error already and never reach this code here if we didn't already find ml64/ml?

Reply:
Yes, it is like this: when find_program is used for ml/ml64.exe it will point to cl.exe (Why? I dont know). As all files are renamed to asm.obj.s, cl.exe will complain about unrecognized file and abort. What cmake has is enable_language(ASM_MASM), which for asm gets ml.exe/ml64.exe depending on build environment. I did exactly the same as CMake to let nmake/jom build properly asm.obj.s files. As this only hits jom/nmake, there are no side effects on other build.

Reply:
Hmm, strange.  It seems like this new code should be moved to right after the find_program code then, with a comment explaining about the finding of cl.exe.

Or you're saying that we could instead have a check for NMake and replace the current enable_language(ASM) on line 182 with enable_language(ASM_MASM)?

Reply:
for enable_language(ASM_MASM) I've tried as a quick fix, and it failed. Preprocessing on asm files wasn't done. Instead of trying to work my way around enable_language(ASM_MASM), and possibly breaking something else, I chose to do what in a nutshell ASM_MASM does, and that is to define CMAKE_ASM_COMPILER as ml.exe or ml64.exe (which is somehow set to be cl.exe) depending on the architecture, and only when "NMake Makefiles" is used. This doesn't happen when building from VisualStudio, and I didn't go into deep understanding why is that so (maybe VisualStudio build properly recognizes .s as .asm).

Reply:
Sure, let's stick with this direct setting then, but I'm suggesting that it should go up where the find_program is for ml64, since that's where the code is trying to find ml64 in the first place, along with a comment about the cl issue as requested.

Reply:
Sure, I will change it to be in the right place.